WordPress/Sniffs/WP/GetMetaFunctionSingleParameterSniff.php

```diff
@@ -0,0 +1,179 @@
+	 *                                                              the relevant parameters.
+	 */
+	protected $target_functions = array(
+		'get_comment_meta'     => array(
+			'condition'   => array(
+				'parameter' => 'key',
+				'position'  => 2,
+			),
+			'recommended' => array(
+				'parameter' => 'single',
+				'position'  => 3,
+			),
+		),
+		'get_metadata'         => array(
+			'condition'   => array(
+				'parameter' => 'meta_key',
+				'position'  => 3,
+			),
+			'recommended' => array(
+				'parameter' => 'single',
+				'position'  => 4,
+			),
+		),
+		'get_metadata_default' => array(
+			'condition'   => array(
+				'parameter' => 'meta_key',
+				'position'  => 3,
+			),
+			'recommended' => array(
+				'parameter' => 'single',
+				'position'  => 4,
+			),
+		),
+		'get_metadata_raw'     => array(
+			'condition'   => array(
+				'parameter' => 'meta_key',
+				'position'  => 3,
+			),
+			'recommended' => array(
+				'parameter' => 'single',
+				'position'  => 4,
+			),
+		),
+		'get_post_meta'        => array(
+			'condition'   => array(
+				'parameter' => 'key',
+				'position'  => 2,
+			),
+			'recommended' => array(
+				'parameter' => 'single',
+				'position'  => 3,
+			),
+		),
+		'get_site_meta'        => array(
+			'condition'   => array(
+				'parameter' => 'key',
+				'position'  => 2,
+			),
+			'recommended' => array(
+				'parameter' => 'single',
+				'position'  => 3,
+			),
+		),
+		'get_term_meta'        => array(
+			'condition'   => array(
+				'parameter' => 'key',
+				'position'  => 2,
+			),
+			'recommended' => array(
+				'parameter' => 'single',
+				'position'  => 3,
+			),
+		),
+		'get_user_meta'        => array(
+			'condition'   => array(
+				'parameter' => 'key',
+				'position'  => 2,
+			),
+			'recommended' => array(
+				'parameter' => 'single',
+				'position'  => 3,
+			),
+		),
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param string $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched
+	 *                                in lowercase.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		$condition   = $this->target_functions[ $matched_content ]['condition'];
+		$recommended = $this->target_functions[ $matched_content ]['recommended'];
+
+		$meta_key = PassedParameters::getParameterFromStack( $parameters, $condition['position'], $condition['parameter'] );
+		if ( false === $meta_key ) {
+			return;
+		}
+
+		$single = PassedParameters::getParameterFromStack( $parameters, $recommended['position'], $recommended['parameter'] );
+		if ( is_array( $single ) ) {
+			return;
+		}
```

Comment:
`getParameterFromStack()` returns either an `array` with the param information or `false` if the parameter was not passed. In the first call to this method, the sniff should bail early if the `$[meta_]key` parameter is not passed to the get meta function. Thus the code is checking if `$meta_key` is false. The second call is only executed if the `$[meta_]key` parameter is passed. In this case, the sniff should bail early if the `$single` parameter is also passed. Thus the code is checking if `$single` is an array.

I have a feeling that in the first version that I created, I was using `false !== $single`. But then opted to use `is_array( $single )` as this seems easier to read to me. It gives more information about what the code expects the `$single` variable to be.

Do you think it would be better to use `! is_array( $meta_key )` for the first check?

Reply:
I should also note that I used `false === $meta_key` instead of `! $meta_key` as the PHPCS coding standards require the former format. I didn't realize that the WPCS coding standards does not require the same format.

Reply:
Yes, I think checking `! is_array()` make sense here and makes it easier to read/understand the code. :+1: